data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/leader/LeaderScheduler.java

```diff
@@ -0,0 +1,172 @@
+                    // will always be a leader until shutdown
+                    coordinator.saveProgressStateForPartition(leaderPartition, Duration.ofMinutes(DEFAULT_EXTEND_LEASE_MINUTES));
+                }
+                try {
+                    Thread.sleep(leaseInterval.toMillis());
+                } catch (final InterruptedException e) {
+                    LOG.info("InterruptedException occurred");
+                    break;
+                }
+            }
+        }
+        // Should Stop
+        LOG.warn("Quitting Leader Scheduler");
+        if (leaderPartition != null) {
+            coordinator.giveUpPartition(leaderPartition);
+        }
+    }
+
+    private boolean isExportRequired(final CollectionConfig.IngestionMode ingestionMode) {
+        return ingestionMode == CollectionConfig.IngestionMode.EXPORT_STREAM ||
+                ingestionMode == CollectionConfig.IngestionMode.EXPORT;
+    }
+
+    private boolean isStreamRequired(final CollectionConfig.IngestionMode ingestionMode) {
+        return ingestionMode == CollectionConfig.IngestionMode.EXPORT_STREAM ||
+                ingestionMode == CollectionConfig.IngestionMode.STREAM;
+    }
+    private void init() {
+        LOG.info("Try to initialize DocumentDB Leader Partition");
+
+        collectionConfigs.forEach(collectionConfig -> {
+            // Create a Global state in the coordination table for the configuration.
+            // Global State here is designed to be able to read whenever needed
+            // So that the jobs can refer to the configuration.
+            coordinator.createPartition(new GlobalState(collectionConfig.getCollection(), null));
+
+            final Instant startTime = Instant.now();
+            final boolean exportRequired = isExportRequired(collectionConfig.getIngestionMode());
+            LOG.info("Ingestion mode {} for Collection {}", collectionConfig.getIngestionMode(), collectionConfig.getCollection());
+            if (exportRequired) {
+                createExportPartition(collectionConfig, startTime);
+            }
+
+            if (isStreamRequired(collectionConfig.getIngestionMode())) {
+                createStreamPartition(collectionConfig, startTime, exportRequired);
+            }
+
+        });
+
+        LOG.debug("Update initialization state");
+        LeaderProgressState leaderProgressState = leaderPartition.getProgressState().get();
+        leaderProgressState.setInitialized(true);
+    }
+
+
+    /**
+     * Create a partition for a stream job in the coordination table.
+     *
+     * @param collectionConfig  collection configuration object containing collection details
+     * @param streamTime the start time for change events, any change events with creation datetime before this should be ignored.
+     */
+    private void createStreamPartition(final CollectionConfig collectionConfig, final Instant streamTime, final boolean waitForExport) {
+        LOG.info("Creating stream global partition");
```

Comment:
Might be helpful to log the collection id here

Reply:
Done